src/blocks/recommendations/class-recommendations-block.php

```diff
@@ -49,41 +49,38 @@ public function run(): void {
 	 *
 	 * @return void
 	 */
-	public static function register_block_and_assets(): void {
-		// Temporary workaround - don't register block when in FSE due to issues.
-		global $pagenow;
-		if ( 'site-editor.php' === $pagenow ) {
-			return;
-		}
+	public static function register_block(): void {
 
-		$plugin_url = plugin_dir_url( PARSELY_FILE );
+		// viewScript was added in WordPress 5.9 so we need to manually register our script if we're < 5.9.
+		// This can be removed if MINIMUM_WORDPRESS_VERSION is set to 5.9.
+		global $wp_version;
 
-		$editor_asset_file = require plugin_dir_path( PARSELY_FILE ) . 'build/recommendations-edit.asset.php';
-		wp_register_script(
-			'wp-parsely-recommendations-block-editor',
-			$plugin_url . 'build/recommendations-edit.js',
-			$editor_asset_file['dependencies'],
-			$editor_asset_file['version'],
-			true
-		);
+		if ( ! isset( $wp_version ) || version_compare( $wp_version, '5.9' ) < 0 ) {
+			$plugin_url      = plugin_dir_url( PARSELY_FILE );
+			$view_asset_path = plugin_dir_path( PARSELY_FILE ) . 'build/blocks/recommendations/view.asset.php';
+			if ( file_exists( $view_asset_path ) ) {
+				$view_asset_file = require plugin_dir_path( PARSELY_FILE ) . 'build/blocks/recommendations/view.asset.php';
+				wp_register_script(
+					'wp-parsely-recommendations-view-script',
+					$plugin_url . 'build/blocks/recommendations/view.js',
+					$view_asset_file['dependencies'],
+					$view_asset_file['version'],
+					true
+				);
+			}
+		}
 
-		$script_asset_file = require plugin_dir_path( PARSELY_FILE ) . 'build/recommendations.asset.php';
-		wp_register_script(
-			'wp-parsely-recommendations-block',
-			$plugin_url . 'build/recommendations.js',
-			$script_asset_file['dependencies'],
-			$script_asset_file['version'],
-			true
-		);
+		/**
+		 * Register the block by passing the path to it's block.json file that contains the majority of it's definition.
+		 * This file will be copied into `build/blocks/recommendations` by the build process and should be accessed there.
+		 *
+		 * @see https://developer.wordpress.org/reference/functions/register_block_type/
+		 */
 
-		wp_register_style(
-			'wp-parsely-recommendations-block',
-			$plugin_url . 'build/style-recommendations-edit.css',
-			array(),
-			$script_asset_file['version']
+		register_block_type(
+			plugin_dir_path( PARSELY_FILE ) . 'build/blocks/recommendations/',
+			self::get_block_registration_args()
```

Comment:
Let's inline the array returned from the `get_blocks_registration_args()` method, and then delete that method.